### PR TITLE
request books

### PR DESCRIPTION
**Pull request by gabrielbursztein**
Now you can request books and see the ones you had requested.

### CODE REVIEW

File: .byebug_history [hunk 1]
```diff
@@ -0,0 +1,33 @@
+exit
```
Comment (santiagovidal) [inline .byebug_history]: Add this file to gitignore.

Remember to refresh the git cache so this file disappears from the PR.

File: app/controllers/books_controller.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@ class BooksController < ApplicationController
     
     def index
       @books = Book.all
-    end
+     end
```
Comment (santiagovidal) [inline app/controllers/books_controller.rb]: Fix this indentation

File: app/controllers/requests_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class RequestsController < ApplicationController
+
+  def index
+    @requests = current_user.requests
+  end
+
+  def Show
```
Comment (santiagovidal) [inline app/controllers/requests_controller.rb]: Delete this method if it isn't used.

Also remember: method names are always in lower case.

File: app/controllers/requests_controller.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+  end
+
+  def new
```
Comment (santiagovidal) [inline app/controllers/requests_controller.rb]: Delete this method.

File: app/controllers/requests_controller.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    @request = Request.create(request_params)
+  end 
+
+  def create
+    book = Book.find(params[:book_id])
+    @request = book.requests.create(user_id: current_user.id)
+    @book_id = params[:book_id]
```
Comment (santiagovidal) [inline app/controllers/requests_controller.rb]: Suggestion: make `book` an instance variable (`@book`) and delete `@book_id`

Remember to use `@book.id` in `create.js.erb`

File: app/views/books/index.html.erb [hunk 6]
```diff
@@ -1,15 +1,30 @@
...
       <td><%= book.author %></td>
+      <td><%= book.year %></td>
       <td><%= book.description %></td>
       <td><%= link_to 'Show', book_path(book) %></td>
+      <% if !current_user.requested?(book) %>
+        <td id='<%= book.id %>'>
+          <%= link_to 'Request', requests_path(:book_id => book.id), method: :post, remote: true %>
+        </td>
+      <% else %>
+        <% case  current_user.request_status(book) %>
```
Comment (santiagovidal) [inline app/views/books/index.html.erb]: You can remove this `case` and just call `request_status(book)`. If you want to make the first letter upper case, you can use the method `titleize`

File: app/views/requests/create.js.erb [hunk 7]
```diff
@@ -0,0 +1 @@
+$('#<%= @book_id %>').html('<p>Pending</p>');
```
Comment (santiagovidal) [inline app/views/requests/create.js.erb]: It's a good practice to not hardcode this sort of thing.

You can call `@request.status.titleize` and you are covered if in the future you want to rename the statuses.

**Review by santiagovidal — DISMISSED** [review]

File: app/models/request.rb [hunk 8]
```diff
@@ -0,0 +1,6 @@
+class Request < ApplicationRecord
+  belongs_to :book
+  belongs_to :user
+  enum status: [:pending, :denied, :approved]
```
Comment (GuilleLeopold) [inline app/models/request.rb]: The `enum` goes before the `belongs_to`. 
There's a style guide, you can check it here https://github.com/bbatsov/rails-style-guide#keep-ar-defaults

**Review by GuilleLeopold — DISMISSED** [review]

**Review by MaicolBen — APPROVED** [review]

**Review by GuilleLeopold — APPROVED** [review]

**Pull request merged**